services/welding-machine-defect-detection-model-service/app/models/model_loader.py

```
@@ -0,0 +1,41 @@
+            f"Scaler file for signal type '{signal_type}' not found")
+    except Exception as e:
+        raise RuntimeError(
+            f"Failed to load scaler for signal type '{signal_type}': {str(e)}")
+
+
+def load_model_file(signal_type: str):
+    try:
+        return load_model(f"app/models/trained_models/welding_{signal_type}_model.keras")
+    except (FileNotFoundError, OSError) as e:
+        raise FileNotFoundError(
+            f"Model file for signal type '{signal_type}' not found: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)